pkg/abstractions/endpoint/task.go

```diff
@@ -74,7 +74,12 @@ func (t *EndpointTask) Cancel(ctx context.Context, reason types.TaskCancellation
 }
 
 func (t *EndpointTask) HeartBeat(ctx context.Context) (bool, error) {
-	heartbeatKey := Keys.endpointRequestHeartbeat(t.msg.WorkspaceName, t.msg.StubId, t.msg.TaskId)
+	task, err := t.es.backendRepo.GetTask(ctx, t.msg.TaskId)
```

Comment:
Downside is that we'll have to retrieve the task, which is not great, but the heartbeat check only happens every 5 seconds right now so if requests are faster than that they'll never hit these check